server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -291,6 +292,13 @@ protected List<LoadedPlugin> plugins() {
         return this.plugins;
     }
 
+    public void registerPluginFeatures(FeatureService service) {
```

Comment:
I don't like this being here, but the ES version in the descriptor is not accessible outside this class

Reply:
I kind of like it instead. Are you worried about something in particular? e.g. inititialization/timing issues?

Reply:
I also like how registerFeatures is implemented. Easy to use.